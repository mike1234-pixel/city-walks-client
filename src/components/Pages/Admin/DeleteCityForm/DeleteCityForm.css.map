{"version":3,"sourceRoot":"","sources":["DeleteCityForm.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;EACA","file":"DeleteCityForm.css"}