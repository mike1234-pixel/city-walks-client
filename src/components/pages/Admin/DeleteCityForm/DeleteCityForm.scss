.delete-city-form {
  padding: 20px;
}
