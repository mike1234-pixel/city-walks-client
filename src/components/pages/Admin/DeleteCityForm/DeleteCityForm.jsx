import { useState } from "react"
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact"
import axios from "axios"
import toTitleCase from '../../../../functions/toTitleCase'
import './DeleteCityForm.css'

const DeleteCityForm = () => {

    const [city, setCity] = useState("")

    const handleChange = (event) => {
        switch(event.target.name) {
            case "city":
              setCity(event.target.value)
              break;
          } 
    }

    const handleSubmit = (event) => {
        event.preventDefault()

        let payload = {
            city: toTitleCase(city)
          };

        axios
        .delete("http://localhost:5000/delete-city",{ data: payload } )
        .then((res, err) => {
          if (err) {
            console.log(err);
          } else if (res.data === "city deleted") {
            console.log("city deleted")
          } else {
            console.log("city not deleted")
          }
        });

        alert("City Deleted")
        setCity("")
        window.scrollTo(0, 0);
    }

    return (
    <div>
        <h2>Delete City</h2>
        <form onSubmit={handleSubmit} className="delete-city-form">
            <MDBInput type="text" name="city" id="city" value={city} label="city name" onChange={handleChange} required/>
            <MDBBtn outline color="elegant" type="submit">
                Delete City <MDBIcon icon="trash"/>
          </MDBBtn>
        </form>
    </div>
    )
}

export default DeleteCityForm