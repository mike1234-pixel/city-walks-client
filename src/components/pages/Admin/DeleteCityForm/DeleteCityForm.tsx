import React, { useState } from "react";
import { MDBInput, MDBBtn, MDBIcon } from "mdbreact";
import axios, { AxiosError, AxiosResponse } from "axios";
import toTitleCase from "../../../../utils/toTitleCase";
import SelectedCity from "../../../../types/PostRequests/SelectedCity";
import "./DeleteCityForm.css";

const DeleteCityForm = () => {

  const [city, setCity] = useState<string>("");

  const handleChange: (event: React.ChangeEvent<any>) => void = (event) => {
    switch (event.target.name) {
      case "city":
        setCity(event.target.value);
        break;
    }
  };

  const handleSubmit: (event: React.FormEvent) => void = (event) => {
    event.preventDefault();

    let payload: SelectedCity = {
      city: toTitleCase(city),
    };

    axios
      .delete("https://city-walks-production.up.railway.app/delete-city", { data: payload })
      .then((res: AxiosResponse) => {
        if (res.data === "city deleted") {
          console.log("city deleted");
        } else {
          console.log("city not deleted");
        }
      })
      .catch((err: AxiosError) => {
        console.log(err);
      });

    alert("City Deleted");
    setCity("");
    window.scrollTo(0, 0);
  };

  return (
    <div>
      <h2>Delete City</h2>
      <form onSubmit={handleSubmit} className='delete-city-form'>
        <MDBInput
          type='text'
          name='city'
          id='city'
          value={city}
          label='city name'
          onChange={handleChange}
          required
        />
        <MDBBtn outline color='elegant' type='submit'>
          Delete City <MDBIcon icon='trash' />
        </MDBBtn>
      </form>
    </div>
  );
};

export default DeleteCityForm;
