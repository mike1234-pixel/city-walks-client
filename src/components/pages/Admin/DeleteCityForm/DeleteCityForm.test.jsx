import renderer from 'react-test-renderer'
import DeleteCityForm from "./DeleteCityForm"

// snapshot test
it('DeleteCityForm Page renders correctly', () => {  

  const tree = renderer
    .create(
        <DeleteCityForm />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});