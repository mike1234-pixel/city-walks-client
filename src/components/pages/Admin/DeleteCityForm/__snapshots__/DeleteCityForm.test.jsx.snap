// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteCityForm Page renders correctly 1`] = `
<div>
  <h2>
    Delete City
  </h2>
  <form
    className="delete-city-form"
    onSubmit={[Function]}
  >
    <div
      className="md-form"
    >
      
      <input
        aria-disabled={false}
        className="form-control"
        data-test="input"
        disabled={false}
        id="city"
        name="city"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onInput={[Function]}
        required={true}
        type="text"
        value=""
      />
      <label
        aria-labelledby=""
        className=""
        data-error=""
        data-success=""
        htmlFor="city"
        id=""
        onClick={[Function]}
      >
        city name
      </label>
    </div>
    <button
      className="btn-elegant btn-outline-elegant btn Ripple-parent"
      data-test="button"
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="submit"
    >
      Delete City 
      <i
        className="fa fa-trash"
        data-test="fa"
      />
      <div
        className="Ripple Ripple-outline "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
  </form>
</div>
`;
